File: app/src/main/java/com/dumuzeyn/mp3player/ValueProviders.java
package com.dumuzeyn.mp3player;

interface ValueProvider<T> {
    T get();
}

interface BooleanValueProvider {
    boolean get();
}

interface IntValueProvider {
    int get();
}

interface TrackFinder {
    Track find(String value);
}

interface TrackPredicate {
    boolean test(Track track);
}

File: app/src/main/java/com/dumuzeyn/mp3player/ValueProviders.kt
package com.dumuzeyn.mp3player

fun interface ValueProvider<T> {
    fun get(): T
}

fun interface BooleanValueProvider {
    fun get(): Boolean
}

fun interface IntValueProvider {
    fun get(): Int
}

fun interface TrackFinder {
    fun find(value: String): Track?
}

fun interface TrackPredicate {
    fun test(track: Track): Boolean
}
